src/focus.service.ts

```diff
@@ -398,7 +392,6 @@ export class FocusService {
       // Trigger focus changes and add selected classes everywhere
       // from the target element to the root.
       for (let el = next; el !== this.root && el; el = el.parentElement) {
-        el.classList.add(cssClass.selected);
         this.triggerFocusChange(el, next);
```

Comment:
calling triggerFocusChange for all the parent elements looks expensive. 
+ we can get the same functionality using focusin and focusout